.github/workflows/lint.yml

```diff
@@ -11,6 +11,8 @@ on:
       - release/*
       - landchecks/*
   workflow_dispatch:
+  schedule:
```

Comment:
This doesn't look right.  I don't think you need this because lint jobs are already running on all main commits atm https://hud.pytorch.org/hud/pytorch/pytorch/22f179d77d9fcd938303cf975ca0c069275e1f29/1?per_page=50&name_filter=lint, so scheduling it again daily doesn't make sense

Reply:
@huydhn how do we make it run not for a PR?
That's needed for a periodic sanity check to catch links that don't work anymore.
And when it runs for a PR it checks the modified files only, not the whole codebase.

Reply:
The way I'm thinking about is to move these 2 check into a separate workflow called `_link_check.yml`.  Then you can:

* Call it from `lint.yml` which runs on every PR, have the flag here to check only for modified files
* Call it from `periodic.yml` which run every 4 hours or `nightly.yml` which run daily on the whole codebase

So, no code duplication, only 2 callers and a share `_link_check.yml`

Reply:
@huydhn updated